File: Stack/stack.py
class Stack():
    MyName = "Gouri Shanker"
    def __init__(self, size=5):
        self.Max_size = size
        self.arr = size * [None]
        self.top = -1

    def push(self, value):
        if value %2 != 0:
            if self.top < self.Max_size - 1:
                self.top += 1
                self.arr[self.top] = value

            else:
                print()
                print("\nStack Overflow!")
        else :
            print("Invalid ,Enter odd value!")

    def pop(self):
        if self.top == -1:
            print("\nStack Underflow!")
        else:
            value = self.arr[self.top]
            self.top -= 1
            print("Pop Sucessful!\n")
            return value**2

    def __str__(self):
        length = self.top
        s = ""
        for i in range(length, -1, -1):
            s = s + str(self.arr[i]) + "\n"
        return s


list = Stack()
print("Enter the operation you want to perform:")
print("Enter 1 to push")
print("Enter 2 to pop")
print("Enter 3 to exit")
while True:
    option = int(input("Enter your choice:"))
    if option == 1:
        list.push(int(input("Enter the value you want to push:")))
        print()
        print(list)
        print()
    if option == 2:
        print(list.pop())
        print()
        print(list)
        print()
    if option == 3:
        break

File: Stack/Stack.java
import java.util.Scanner;
public class Stack{
	int size=5,top=-1;
	int arr[]=new int[size];

	public void push(int value){
		if(top==size-1){
			System.out.println("\n        Stack Overflow.!\n");
		}
		else{
			top++;
			arr[top]=value;
		}
	}

	public int pop(){
		int x=0;
		if(top==-1){
			System.out.println("\n      Stack Underflow.!\n");
			return x;
		}
		else{
			x=arr[top];
			top--;
			return x;
		}
	}


	public String toString(){
    String s ="\n";
		if(top==-1){
			s = "       \nStack is Empty!\n";
		}
		else{
			for(int i=top;i>=0;i--)
			{
			  s+= Integer.toString(arr[i])+"\n" ;
		}

		}
		return s;
	}

	public static void main(String[] args) {
		Stack list = new Stack();
		Scanner in = new Scanner(System.in);
		System.out.println("Enter the operation you want to perform:");
		System.out.println("Enter 1 to push");
		System.out.println("Enter 2 to pop");
		System.out.println("Enter 3 to exit");

		while (true){

		    System.out.print("Enter your choice:");
			  int	option = in.nextInt();
		    if (option == 1)
				{
				    System.out.print("Enter the value you want to push:");
				    int value = in.nextInt();
		        list.push(value);
		        System.out.println(list);}
		    else if (option == 2){
		        list.pop();
		        System.out.println(list);
					}
		    else if (option == 3){
		        break;
					}
				else{
					System.out.println("Invalid input :(");
				}
	}
	Stack list1 = new Stack();
	System.out.println(list1);
}
}
